explorer/interpreter/value.h

```diff
@@ -145,19 +145,113 @@ auto ValueEqual(Nonnull<const Value*> v1, Nonnull<const Value*> v2,
                 std::optional<Nonnull<const EqualityContext*>> equality_ctx)
     -> bool;
 
+class SizedInteger {
```

Comment:
Instead of a base class and a bunch of derived classes, could this be a class _template_ (or maybe a pair of class templates, one for signed and one for unsigned) that takes the number of bytes as a template parameter? That seems like it would result in less code duplication.

Reply:
👍 Although if i may ask, could you elaborate a bit more on the technical details on how the template type should be the number of bytes and how that translates into the actual datatype?
> takes the number of bytes as a template parameter

in not entirely sure if i understood the technical requirement here correctly.
To my understanding the idea is to do somethink like:
```cc
SizedInteger<uint8_t>... 
SizedInteger<int16_t>... 
```
but that does not align to the number of bytes as its directly passing the type and not a byte size, correct?